specs/_features/eip7594/fork-choice.md

```diff
@@ -0,0 +1,187 @@
+# EIP-7594 -- Fork Choice
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+  - [Helpers](#helpers)
+    - [Modified `is_data_available`](#modified-is_data_available)
+    - [New `is_chain_available`](#new-is_chain_available)
+    - [Modified `get_head`](#modified-get_head)
+    - [New `is_peer_sampling_required`](#new-is_peer_sampling_required)
+- [Updated fork-choice handlers](#updated-fork-choice-handlers)
+  - [`on_block`](#on_block)
+    - [Pull-up tip helpers](#pull-up-tip-helpers)
+      - [`compute_pulled_up_tip`](#compute_pulled_up_tip)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the modification of the fork choice accompanying EIP-7594.
+
+### Helpers
+
+#### Modified `is_data_available`
+
+```python
+def is_data_available(beacon_block_root: Root, require_peer_sampling: bool=False) -> bool:
+    column_sidecars = retrieve_column_sidecars(beacon_block_root, require_peer_sampling)
```

Comment:
We still need to add a comment to this saying `retrieve_column_sidecars` a fake function.

Maybe something like the following:

```
    # `retrieve_column_sidecars` is implementation and context dependent. It returns `require_peer_sampling` column
    # sidecars for the given `beacon_block_root` or raises an exception if they are not available. The p2p network
    # does not guarantee sidecar retrieval outside of `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS` epochs.
```